var/spack/repos/builtin/packages/python/package.py

```diff
@@ -195,6 +195,7 @@ class Python(AutotoolsPackage):
     patch('python-3.6.8-distutils-C++.patch', when='@3.6.8,3.7.2')
     patch('python-3.7.3-distutils-C++.patch', when='@3.7.3')
     patch('python-3.7.4+-distutils-C++.patch', when='@3.7.4:')
```

Comment:
Shouldn't this also be done for 2.7.8 to 3.7.3?

Reply:
Strictly said: No, because not all of these releases (2.7.8 to 3.7.3) have a distutils-C++ patch.

In practicality, it is not really worth to write testsuite patches for each of these individual versions, since none of them pass the rest of the python testsuite anyways in spack. The other testsuite fails already start coming when going just back to 3.7.9.

Reply:
@bernhardkaindl: Doing my best to recall my headspace when working on #17140, I recall throwing some of these patches together without the greatest deal of precision. Based on the PR comments, it looks like I took these patches more-or-less verbatim from [a patch set in the Python bug tracker](https://bugs.python.org/issue1222585). The patch set was never merged, so it doesn't surprise me that some of its constituents are imperfect. It looks like you were able to diagnose these imperfections and come up with a solution though, so thanks for looking into this and kudos!

Reply:
Thanks, I appreciate that!

I wanted to see if I could get `spack install --test all` working for the core packages, so I found many things in the last days.

I'm not surprised of such 'gotchas' anymore after that, and as these patches didn't even update the `distutils` testsuite to pass, it's no surprise that some issues were missed when those patches were developed.

Reply:
This is just a little be of archealogy:

Gentoo carried these patches until March (found in the git log after git cloning https://cgit.gentoo.org/repo/gentoo.git).

In 2015, Erik Hvatum (using gentoo) found the same issue already and two years later Josef Kemetmüller applied Erik's fix, but only to the patch for python3.4, but he didn't update the python2.7 patch, so that's were the lines of the patches diverged. That means the patch for "3.x" from Josef has the cflags fix since 2017:

https://bugs.python.org/issue1222585#msg242706

Erik linked the gentoo bug report and these patches originate from gentoo. Mike Gilbert didn't know the reason why they wiped out the cflags, it was before his time: https://bugs.gentoo.org/548776#c3

He know it and was pretty sure it was intentional, but the reason might have been anything: Maybe they didn't want the `CFLAGS` of python itself used for module builds as well, but that is actually the intention of the python devs in doing so.

It should be safe to do what upstream Python decided to do.

@xjrc Thanks for approving the fix.